# 9.4 검정과 유의확률

### 연습문제 9.4.1

1. 베르누이 분포
2. Ho: μ =0.5 / Hc: μ > 0.5 (μ: 상품평이 '좋아요'일 확률)

### 연습문제 9.4.2

1. '좋아요' 상품평의 수 / '좋아요' + '싫어요' 상품형의 수
2. 이항분포

### 연습문제 9.4.3

In [2]:
rv = sp.stats.binom(20, 0.5)
1 - rv.cdf(10)

0.41190147399902344

유의확률이 약 0.412로 , 유의수준인 10%보다 크다. 즉, 이 상품이 보통 수준이라는 귀무가설을 기각할 수 없다.

# 9.5 사이파이를 사용한 검정

### 연습문제 9.5.1

1.

In [5]:
for i in range(0, 11):
    print(sp.stats.binom_test(i,10,0.5))

0.001953125
0.021484375
0.109375
0.34375
0.75390625
1.0
0.75390625
0.34375
0.109375
0.021484375
0.001953125


1,2,9,10번

2.

In [6]:
for i in range(0, 1001):
    value = sp.stats.binom_test(i,1000,0.5)
    if value < 0.1:
        print(i)

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199
200
201
202
203
204
205
206
207
208
209
210
211
212
213
214
215
216
217
218
219
220
221
222
223
224
225
226
227
228
229
230
231
232
233
234
235
236
237
238
239
240
241
242
243
244
245
246
247
248
249
250
251
252
253
254
255
256
257
258
259
260
261
262
263
264
265
266
267
268
269
270
271
272
273
274
275
276
27

0-473번/ 527-1000번

In [8]:
474+(1000-527+1)
print('n=10인경우 공정하지 않다고 이야기할 수 있는 횟수의 비율:',0.4)
print('n=1000인경우 공정하지 않다고 이야기할 수 있는 횟수의 비율:', 948/1000 )

n=10인경우 공정하지 않다고 이야기할 수 있는 횟수의 비율: 0.4
n=1000인경우 공정하지 않다고 이야기할 수 있는 횟수의 비율: 0.948


### 연습문제 9.5.2

1.

In [10]:
import seaborn as sns
tips = sns.load_dataset('tips')

In [11]:
tips

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

In [14]:
female_smoker = tips[tips.sex == 'Female'].smoker.value_counts() 
female_smoker

No     54
Yes    33
Name: smoker, dtype: int64

In [18]:
sp.stats.binom_test(54,87,0.5,'greater')

0.015709055116923054

유의확률인 0.015가 유의 수준인 0.1보다 낮으므로 여성 비흡연자와 여성 흡연자의 비율이 같다는 귀무가설을 기각한다.
즉, 여자 손님 중 비흡연자가 흡연자보다 많다고 할 수 있다.

2.

In [19]:
dinner_female_smoker = tips[(tips.sex == 'Female') & (tips.time == 'Dinner')].smoker.value_counts() 
dinner_female_smoker

No     29
Yes    23
Name: smoker, dtype: int64

In [20]:
sp.stats.binom_test(29,52,0.5,'greater')

0.24422783468994602

유의확률인 0.24가 유의 수준인 0.1보다 높으므로 여성 비흡연자와 여성 흡연자의 비율이 같다는 귀무가설을 기각할 수 없다..
즉, 저녁 시간 대 여자 손님 중 비흡연자가 흡연자보다 많다고 할 수 없다.

### 연습문제 9.5.3

In [39]:
for n in range(1,1500):
    value = sp.stats.binom_test(int(round(n*0.7)),n,2/3,'greater')
    if value < 0.01:
        print('n:', n, 'value:', value)
    break

In [37]:
value = sp.stats.binom_test(1100*0.7,1100,2/3,'greater')
value

0.009906572889331722

1100명